src/wazuh_qa_framework/system/host_manager.py

```
@@ -640,3 +642,36 @@ def append_block_in_file(self, host, path, block, become=None, ignore_errors=Fal
                 raise Exception(f"Error inserting a block in file {path} on host {host}: {result}")
 
         return result
+
+    def run_playbook(self, playbook_path, extra_vars=None):
+        """Run playbook on a host.
+
+        Args:
+            playbook_path (str): Playbook path to run.
+            path (str): The full path of the file/object to get the facts of.
+            extra_vars (dict): Extra variables to be passed to Ansible at runtime
+
+        Returns:
+            bool: Runner result.
+
+        Raises:
+            Exception: If the playbook fails.
+        """
+        inventory_dict = yaml.safe_load(open(self.inventory_path))
+
+        try:
+            runner = ansible_runner.run(
+                playbook=playbook_path,
+                inventory=inventory_dict,
+                extravars=extra_vars,
+            )
+            if runner.status == 'failed':
+                raise Exception(f"Run playbook has status failed")
+
+            if runner.status == 'timeout':
+                raise Exception(f"The playbook execution has failed due to a tiemout")
+
+        except Exception as e:
+            raise Exception(f"Run playbook fails due to an unexpected error: {e}")
+
+        return runner.rc == 0
```

Is necessary this return?

---

I think it's ok to know if the playbook runs successfully. Why do you think it's not necessary?

---

Because if it fails it should raise an Exception, so the return statement does not provide extra information.
IMHO is better to return the runner object instead to gather stdout, stderr, events, etc

---

Done 20f7457ab10f4e94d38862d944ed019fe74da734

---

Done 20f7457ab10f4e94d38862d944ed019fe74da734